### ISSUE TITLE
broker1 test is flaky

### ISSUE THREAD

**Issue by rgal** [SNSystems/pstore#54]
Seen failing sporadically on Linux and Windows.

On Linux (https://travis-ci.org/SNSystems/pstore/jobs/650492100):

```
FAIL: PStore :: broker/broker1.test (19 of 19)
******************** TEST 'PStore :: broker/broker1.test' FAILED ********************
Script:
--
: 'RUN: at line 10';   /usr/bin/python "/home/travis/build/SNSystems/pstore/system_tests/broker/broker1.py" "/home/travis/build/SNSystems/pstore/build/bin" > "/home/travis/build/SNSystems/pstore/build/bin/system_tests/broker/Output/broker1_actual.txt"
: 'RUN: at line 11';   diff "/home/travis/build/SNSystems/pstore/build/bin/system_tests/broker/Output/broker1_actual.txt" "/home/travis/build/SNSystems/pstore/system_tests/broker/broker1_expected.txt"
--
Exit Code: 1
Command Output (stderr):
--
+ : 'RUN: at line 10'
+ /usr/bin/python /home/travis/build/SNSystems/pstore/system_tests/broker/broker1.py /home/travis/build/SNSystems/pstore/build/bin
Running broker: ['/home/travis/build/SNSystems/pstore/build/bin/pstore-brokerd', '--pipe-path', '/tmp/broker1.pipe']
Running poker: ['/home/travis/build/SNSystems/pstore/build/bin/pstore-broker-poker', '--flood', '100', '--kill', '--pipe-path', '/tmp/broker1.pipe']
/home/travis/build/SNSystems/pstore/system_tests/broker/broker1.py: waiting for poker to exit
/home/travis/build/SNSystems/pstore/system_tests/broker/broker1.py: waiting for broker to exit
pstore broker is exiting.
/home/travis/build/SNSystems/pstore/system_tests/broker/broker1.py: done
/home/travis/build/SNSystems/pstore/system_tests/broker/broker1.py: broker timeout
/home/travis/build/SNSystems/pstore/system_tests/broker/broker1.py: broker exception: Command '['/home/travis/build/SNSystems/pstore/build/bin/pstore-brokerd', '--pipe-path', '/tmp/broker1.pipe']' returned non-zero exit status -9
--
```

On Windows (https://travis-ci.org/SNSystems/pstore/jobs/650492110):
```
  FAIL: PStore :: broker/broker1.test (19 of 19)
  ******************** TEST 'PStore :: broker/broker1.test' FAILED ********************
  Script:
  --
  : 'RUN: at line 10';   C:/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/opt/bin/python.exe "C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1.py" "C:/Users/travis/build/SNSystems/pstore/build/bin/Release" > "C:\Users\travis\build\SNSystems\pstore\build\bin\Release\system_tests\broker\Output/broker1_actual.txt"
  : 'RUN: at line 11';   diff "C:\Users\travis\build\SNSystems\pstore\build\bin\Release\system_tests\broker\Output/broker1_actual.txt" "C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1_expected.txt"
  --
  Exit Code: 1
  
  Command Output (stdout):
  --
  $ ":" "RUN: at line 10"
  $ "C:/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/opt/bin/python.exe" "C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1.py" "C:/Users/travis/build/SNSystems/pstore/build/bin/Release"
  # redirected output from 'C:\\Users\\travis\\build\\SNSystems\\pstore\\build\\bin\\Release\\system_tests\\broker\\Output/broker1_actual.txt':
  poker exited successfully
  ECHO:0
  ECHO:01
  ECHO:012
  ECHO:0123
  ECHO:01234
  ECHO:012345
  ECHO:0123456
  ECHO:01234567
  ECHO:012345678
  ECHO:0123456789
  ECHO:01234567890
  ECHO:012345678901
  ECHO:0123456789012
  ECHO:01234567890123
  ECHO:012345678901234
  ECHO:0123456789012345
  ECHO:01234567890123456
  ECHO:012345678901234567
  ECHO:0123456789012345678
  ECHO:01234567890123456789
  ECHO:012345678901234567890
  ECHO:0123456789012345678901
  ECHO:01234567890123456789012
  ECHO:012345678901234567890123
  ECHO:0123456789012345678901234
  ECHO:01234567890123456789012345
  ECHO:012345678901234567890123456
  ECHO:0123456789012345678901234567
  ECHO:01234567890123456789012345678
  ECHO:012345678901234567890123456789
  ECHO:0123456789012345678901234567890
  ECHO:01234567890123456789012345678901
  ECHO:012345678901234567890123456789012
  ECHO:0123456789012345678901234567890123
  ECHO:01234567890123456789012345678901234
  ECHO:012345678901234567890123456789012345
  ECHO:0123456789012345678901234567890123456
  ECHO:01234567890123456789012345678901234567
  ECHO:012345678901234567890123
  ...
  note: data was truncated
  
  # command stderr:
  Running broker: ['C:/Users/travis/build/SNSystems/pstore/build/bin/Release\\pstore-brokerd.exe', '--pipe-path', '\\\\.\\pipe\\broker1.pipe']
  
  Running poker: ['C:/Users/travis/build/SNSystems/pstore/build/bin/Release\\pstore-broker-poker.exe', '--flood', '100', '--kill', '--pipe-path', '\\\\.\\pipe\\broker1.pipe']
  
  C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1.py: waiting for poker to exit
  
  2020-02-14T16:00:34-0000 - main - notice - broker starting
  
  2020-02-14T16:00:34-0000 - main - notice - opening pipe
  
  2020-02-14T16:00:34-0000 - main - notice - starting threads
  
  2020-02-14T16:00:34-0000 - command - info - Waiting for commands
  
  2020-02-14T16:00:34-0000 - gcwatch - info - starting gc process watch thread
  
  2020-02-14T16:00:34-0000 - gcwatch - info - waiting for a GC process to complete
  
  2020-02-14T16:00:34-0000 - http - info - initializing on port: 8080
  
  2020-02-14T16:00:34-0000 - http - info - starting server-loop
  
  2020-02-14T16:00:34-0000 - uptime - info - uptime 1 second tick starting
  
  2020-02-14T16:00:34-0000 - read0 - notice - listening to named pipe "\\.\pipe\broker1.pipe"
  
  2020-02-14T16:00:34-0000 - read1 - notice - listening to named pipe "\\.\pipe\broker1.pipe"
  
  2020-02-14T16:00:34-0000 - main - notice - waiting
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678901
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678901234
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789012
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789012345
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678901234567
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789012345678
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678901234567890
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789012345678901
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678901234567890123
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789012345678901234
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678901234567890123456
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789012345678901234567
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:012345678901234567890123456789
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:0123456789012345678901234567890
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read0 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read1 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - read1 - debug - Pipe was broken
  
  2020-02-14T16:00:34-0000 - command - info - verb:ECHO path:01234567890123456789012345678901...
  
  2020-02-14T16:00:34-0000 - read0 - debug - Queueing command
  
  2020-02-14T16:00:34-0000 - command - info - verb:SUICIDE path:
  
  pstore broker is exiting.
  
  2020-02-14T16:00:34-0000 - command - info - performing shutdown
  
  2020-02-14T16:00:34-0000 - command - info - asking gc process watch thread to exit
  
  2020-02-14T16:00:34-0000 - command - info - push command "_QUIT"
  
  2020-02-14T16:00:34-0000 - command - info - push command "_QUIT"
  
  2020-02-14T16:00:34-0000 - command - info - push command "_CQUIT"
  
  2020-02-14T16:00:34-0000 - command - info - Connecting
  
  2020-02-14T16:00:34-0000 - command - info - Connected
  
  2020-02-14T16:00:34-0000 - command - info - shutdown requests complete
  
  2020-02-14T16:00:34-0000 - command - info - verb:_QUIT path:
  
  2020-02-14T16:00:34-0000 - command - info - waking one reader thread
  
  2020-02-14T16:00:34-0000 - command - info - verb:_QUIT path:
  
  2020-02-14T16:00:34-0000 - command - info - waking one reader thread
  
  2020-02-14T16:00:34-0000 - command - info - verb:_CQUIT path:
  
  2020-02-14T16:00:34-0000 - command - info - Exiting command thread
  
  2020-02-14T16:00:34-0000 - read1 - notice - exiting read loop
  
  2020-02-14T16:00:34-0000 - read0 - notice - exiting read loop
  
  2020-02-14T16:00:34-0000 - gcwatch - info - cleaning up
  
  2020-02-14T16:00:34-0000 - scavenger - info - begin scavenging
  
  2020-02-14T16:00:34-0000 - scavenger - info - scavenger thread exiting
  
  C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1.py: waiting for broker to exit
  
  2020-02-14T16:00:35-0000 - uptime - info - uptime thread exiting
  
  C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1.py: done
  
  C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1.py: broker timeout
  
  C:\Users\travis\build\SNSystems\pstore\system_tests\broker/broker1.py: broker exception: Command '['C:/Users/travis/build/SNSystems/pstore/build/bin/Release\\pstore-brokerd.exe', '--pipe-path', '\\\\.\\pipe\\broker1.pipe']' returned non-zero exit status 1
  
  
CUSTOMBUILD : error : command failed with exit status: 1 [C:\Users\travis\build\SNSystems\pstore\build\pstore-system-tests.vcxproj]
  
  --
```

**Comment by paulhuggett** [comment]
This is a problem that I know about and have made several attempts to fix. My current theory is that the quit message from pstore-broker-poker may be occasionally processed before all of the "echo" messages have been executed. It has improved almost impossible to reproduce.

**Comment by paulhuggett** [comment]
> The SIGINT (or Windows equivalent) is received immediately and queues the QUIT message.

This isn't quite true. ATM, the broker-poker sends a “suicide” message to tell the broker to exit. This should probably be changed to using SIGINT as described above since this is a legitimate way that the user may use to kill the process.

**Comment by paulhuggett** [comment]
It’s not a proper fix, but commit ab6daa996c0622db15ba8e7ca3f487bbc1e96854 should reduce the number of build failures that we see.

**Comment by paulhuggett** [comment]
This test hasn’t failed in a long time now. Closing.
